### PR TITLE
Tighten the policy to pull images when creating a container

### PR DESCRIPTION
**Pull request by markusthoemmes**
String checking is bad, type-checking is good!

### CODE REVIEW

**Comment by lehoanganh** [comment]
it means, for each invoking docker action, invoker will try to pull the container, and doesn't care if the image is there locally or not?

**Comment by markusthoemmes** [comment]
Yes, that's semantically preserving what it did before. Docker will fast forward though, and almost immediatly notice that the layers are already there.

**Comment by lehoanganh** [comment]
2 seconds for the check ;)

**Comment by fxulusoy** [comment]
Checking the image manually whether it is there or not does cost some time as well :)

**Comment by markusthoemmes** [comment]
@lehoanganh because an explicit `pull` at least checks, if the latest version is around.

File: core/dispatcher/src/main/scala/whisk/core/container/ContainerPool.scala [hunk 1]
```diff
@@ -483,19 +484,18 @@ class ContainerPool(
         val key = makeKey(action, auth)
         val warmedContainer = if (limits.memory == defaultMemoryLimit && imageName == nodeImageName) getWarmNodejsContainer(key) else None
         val containerName = makeContainerName(action)
-        val con = warmedContainer getOrElse makeGeneralContainer(key, containerName, imageName, limits)
+        val con = warmedContainer getOrElse makeGeneralContainer(key, containerName, imageName, limits, action.exec.isInstanceOf[BlackBoxExec])
```
Comment (cbickel) [inline core/dispatcher/src/main/scala/whisk/core/container/ContainerPool.scala]: Not sure what's cheaper, but `action.exec.kind == Exec.blackbox` would be another solution.

**Comment by jasonpet** [comment]
see pg2 9

**Comment by rabbah** [comment]
Squashed and merged as https://github.com/openwhisk/openwhisk/compare/429a53b7a2e5...60d1bad5ea0b.

**Pull request closed without merging**